### PR TITLE
changed infiniband tests according to configure_network module changed

### PR DESCRIPTION
**Pull request by smruti77**
changed infiniband test according to module hosts.py and interfaces.py.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[mckey.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401042/mckey.log)
[ping6.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401043/ping6.log)
[pingpong.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401044/pingpong.log)
[rdma.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401045/rdma.log)
[rping.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401046/rping.log)
[ucmatose.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401047/ucmatose.log)
[udaddy.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4401048/udaddy.log)

File: io/net/infiniband/mckey.py [hunk 1]
```diff
@@ -72,12 +72,23 @@ def setUp(self):
...
         else:
-            configure_network.set_ip(self.ipaddr, self.netmask, self.iface,
-                                     interface_type='Ethernet')
+            self.networkinterface = NetworkInterface(self.iface, local)
+            try:
+                self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            except:
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            self.networkinterface.bring_up()
```
Comment (manvanthar) [inline io/net/infiniband/mckey.py]: This bringup should be outside of if condition right?  If not it works only for ROCE and not for IB

Comment (smruti77) [reply io/net/infiniband/mckey.py]: @manvanthar  yes, i will change that.

File: io/net/infiniband/rdma_tests.py [hunk 2]
```diff
@@ -61,12 +61,23 @@ def setUp(self):
...
         else:
-            configure_network.set_ip(self.ipaddr, self.netmask, self.iface,
-                                     interface_type='Ethernet')
+            self.networkinterface = NetworkInterface(self.iface, local)
+            try:
+                self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            except:
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            self.networkinterface.bring_up()
```
Comment (manvanthar) [inline io/net/infiniband/rdma_tests.py]: outside if condition

File: io/net/infiniband/ucmatose.py [hunk 3]
```diff
@@ -148,11 +161,11 @@ def tearDown(self):
...
         unset ip
         """
-        if not configure_network.set_mtu_host(self.iface, '1500'):
+        if self.networkinterface.set_mtu('1500') is not None:
             self.fail("Failed to set mtu in host")
-        if not self.peerinfo.set_mtu_peer(self.peer_interface, '1500'):
+        if self.peer_networkinterface.set_mtu('1500') is not None:
             self.fail("Failed to set mtu in peer")
-        configure_network.unset_ip(self.iface)
+        self.networkinterface.remove_ipaddr(self.ipaddr, self.netmask)
```
Comment (manvanthar) [inline io/net/infiniband/ucmatose.py]: Do not unset the IP.  Let the ip be pingable for the cfg tests.

File: io/net/infiniband/udaddy.py [hunk 4]
```diff
@@ -147,11 +160,11 @@ def tearDown(self):
...
         unset ip
         """
-        if not configure_network.set_mtu_host(self.iface, '1500'):
+        if self.networkinterface.set_mtu('1500') is not None:
             self.fail("Failed to set mtu in host")
-        if not self.peerinfo.set_mtu_peer(self.peer_interface, '1500'):
+        if self.peer_networkinterface.set_mtu('1500') is not None:
             self.fail("Failed to set mtu in peer")
-        configure_network.unset_ip(self.iface)
+        self.networkinterface.remove_ipaddr(self.ipaddr, self.netmask)
```
Comment (manvanthar) [inline io/net/infiniband/udaddy.py]: same here this tear doen should only set it back to 1500. IP should not be unset,

File: io/net/infiniband/ping6.py [hunk 5]
```diff
@@ -74,12 +74,23 @@ def setUp(self):
...
         else:
-            configure_network.set_ip(self.ipaddr, self.netmask, self.iface,
-                                     interface_type='Ethernet')
+            self.networkinterface = NetworkInterface(self.iface, local)
+            try:
+                self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            except:
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            self.networkinterface.bring_up()
```
[the remark this answers is not in the document]
Comment (smruti77) [reply io/net/infiniband/ping6.py]: this one done

**Comment by manvanthar** [comment]
Reviewed the logs the tests are either failing or interrupted with control C.   Would like to see the complete running logs.

File: io/net/infiniband/ucmatose.py (hunk 3 above)
Comment (smruti77) [reply io/net/infiniband/ucmatose.py]: @manvanthar for all test cases ..?

**Comment by smruti77** [comment]
> Reviewed the logs the tests are either failing or interrupted with control C. Would like to see the complete running logs.

@manvanthar 2 test are failing and more test are their so cancel after few test.i need to check work flow thats why.

**Comment by smruti77** [comment]
[mckey.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412837/mckey.log)
[ping6.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412838/ping6.log)
[pingpong.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412839/pingpong.log)
[rdma_test.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412840/rdma_test.log)
[rping1.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412841/rping1.log)
[rping2.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412843/rping2.log)
[ucmatose.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412845/ucmatose.log)
[udaddy.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4412846/udaddy.log)

**Comment by PraveenPenguin** [comment]
@smruti77 can you please resolve Travis issue

**Comment by manvanthar** [comment]
@smruti77 i reviewed the code and logs.     So the failures are from ipv6 related options which were working before.  Anyways i will look at it and fix them when i get time.

No further comments from me. I think if Travis issue is fixed , i think its good to go.

@narasimhan-v ^

**Comment by smruti77** [comment]
> @smruti77 i reviewed the code and logs. So the failures are from ipv6 related options which were working before. Anyways i will look at it and fix them when i get time.
> 
> No further comments from me. I think if Travis issue is fixed , i think its good to go.
> 
> @narasimhan-v ^

@manvanthar ok

**Comment by smruti77** [comment]
> @smruti77 can you please resolve Travis issue

ok

**Comment by narasimhan-v** [comment]
@manvanthar thanks for the review.

**Pull request merged**